types/knockout/index.d.ts

```diff
@@ -174,6 +181,23 @@ interface KnockoutObservableArrayStatic {
     <T>(value?: T[] | null): KnockoutObservableArray<T>;
 }
 
+/**
+ * While all observable arrays are writable at runtime, this type is analogous to the native ReadonlyArray type:
+ * casting an observable array to this type expresses the intention that it shouldn't be mutated.
+ */
+interface KnockoutReadonlyObservableArray<T> extends KnockoutReadonlyObservable<ReadonlyArray<T>>, KnockoutReadonlyObservableArrayFunctions<T> {
+    // NOTE: Keep in sync with KnockoutObservableArray<T>, see note on KnockoutObservableArray<T>
+    subscribe(callback: (newValue: KnockoutArrayChange<T>[]) => void, target: any, event: "arrayChange"): KnockoutSubscription;
+    subscribe(callback: (newValue: T[]) => void, target: any, event: "beforeChange"): KnockoutSubscription;
+    subscribe(callback: (newValue: T[]) => void, target?: any, event?: "change"): KnockoutSubscription;
+    subscribe<TEvent>(callback: (newValue: TEvent) => void, target: any, event: string): KnockoutSubscription;
```

Comment:
The TEvent type doesn't seem to be used anywhere. Can it be removed?
```suggestion
    subscribe(callback: (newValue: any) => void, target: any, event: string): KnockoutSubscription;
```

Reply:
I think I agree with you in theory, this is kinda a useless type param; but I'm not adding these types here, I'm just copying them from `KnockoutObservableArray<T>` and it'd be a breaking change to just drop the type param.  

Perhaps this is best left for a separate PR?  And it'd also be wise to make the corresponding changes in the typings for the KO 3.5 RC, since this type is [largely the same there](https://github.com/knockout/knockout/blob/master/build/types/knockout.d.ts#L22).

Reply:
I support the idea to remove that overload.